File: thi-thuc-hanh/src/app/models/car.ts
import {StartPoint} from './start-point';

export interface Car {
  id: number;
  type: string;
  name: string;
  startPoint: StartPoint;
  endPoint: StartPoint;
  phone: string;
  email: string;
  startTime: string;
  endTime: string;
}

File: vehicle-management/src/main/java/com/codegym/model/Car.java
package com.codegym.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "car")
public class Car {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy= GenerationType.AUTO)
    private String id;
    private String type;
    private String name;
    private String phone;
    private String email;
    private String startTime;
    private String endTime;
    @ManyToOne
    @JoinColumn(name = "start_point", referencedColumnName ="id")
    private StartPoint startPoint;
    @ManyToOne
    @JoinColumn(name = "end_point", referencedColumnName ="id")
    private EndPoint endPoint;
}
